pkg/flock/flock_unix.go

```diff
@@ -52,7 +52,7 @@ func Release(lock int) error {
 
 // CheckLock checks whether any process is using the lock
 func CheckLock(path string) bool {
-	lockByte, _ := exec.Command("lsof", "-w", "-F", "ln", path).Output()
+	lockByte, _ := exec.Command("lsof", "-w", "-F", "lfn", path).Output()
```

Comment:
eugh, I wonder if we've ever checked to see if there is a pure go version of this

Reply:
Its literally only used in the unit test itself to validate the lock. The major lock libraries like https://github.com/gofrs/flock don't have a "checkLock" function

Reply:
ah. It is a little weird that we have stuff that is only used by the test framework mixed into the main package; if we only need it in tests maybe it could go in the frameworks stuff. Just nitpicking though.

Reply:
That said, wouldn't `TryLock()` be equivalent to what we're doing with `CheckLock()`?

Reply:
TryLock actually grabs the lock if it can get it. But your right, at a minimum I can move this function into the test file itself, since its only used there. 

| // TryLock is the preferred function for taking an exclusive file lock.
| // This function takes an RW-mutex lock before it tries to lock the file,
| // so there is the possibility that this function may block for a short time
| // if another goroutine is trying to take any action.
| // The actual file lock is non-blocking.
| // If we are unable to get the exclusive file lock,
| // the function will return false instead of waiting for the lock.